src/elexclarity/formatters/base.py

```diff
@@ -1,15 +1,20 @@
 from slugify import slugify
 from dateutil import parser, tz
 
-from elexclarity.formatters.const import STATE_OFFICE_ID_MAPS
+from elexclarity.formatters.const import STATE_OFFICE_ID_MAPS, STATE_RACE_TYPE_MAPS
 
 
 class ClarityConverter(object):
     def __init__(self, statepostal, county_lookup=None, **kwargs):
         self.state_postal = statepostal
         self.county_lookup = county_lookup
 
-    def get_race_type(self, election_name):
+    def get_race_type(self, election_name, contest):
+        contest_name = contest['text']
+        lookup = STATE_RACE_TYPE_MAPS.get(self.state_postal, {})
+        for key, value in lookup.items():
+            if key in election_name or key in contest_name:
```

Comment:
can we set the map to be the contest_name directly instead of the party suffix?

Reply:
Sure. Do you want I should get rid of the ability to match substrings entirely (which I'm also using for "Primary" in the election name), or should we just change what's in the mapping to be the whole contest name?

Reply:
I like the first part (looking for "primary" in the election name) since we do the same for general elections, but I would rather the `STATE_RACE_TYPE_MAPS` be more explicit per contest name -> party in the same way that the `STATE_OFFICE_ID_MAPS` constant is

Reply:
Okay, made those [more explicit](https://github.com/washingtonpost/elex-clarity/pull/18/commits/15c49a90bd0ca4cc2ae74211e497f9cca479f087). Also fixed some test errors in the same commit.

Reply:
One note that tests surfaced: the settings converter doesn't have the contest name, so it goes off the election name and it will generate an ID with P instead of D or R in the race type.